src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -22,6 +23,459 @@ namespace System
     /// </summary>
     public static partial class MemoryExtensions
     {
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a specified element.
+        /// </summary>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static Memory<T> Trim<T>(this Memory<T> memory, T trimElement)
+            where T : IEquatable<T>
+        {
+            int start = ClampStart(memory.Span, trimElement);
+            int length = ClampEnd(memory.Span, start, trimElement);
```

Comment:
I'd expect this to perform better if it accessed .Span once and stored it into a local, in particular if the memory wraps a MemoryManager, where .Span makes a virtual call.